### ISSUE TITLE
Terminal disconnected when another workspace is starting or stopping 

### ISSUE THREAD

**Issue by dorinburcea** [eclipse-che/che#12928]
Hi I am experiencing frequent situation when teminal is disconnected It happens anytime a workspace is stopped or started
I am using the version CHE 6.19.0 in kubernetes with nginx controller. 
Is there any configuration I can do ?
I've seen similar issue here https://github.com/eclipse/che/issues/9943 . 

I am using ingress controller I just want to know if there is a solution for the architecture using nginx controller

**Comment by monaka** [comment]
In my experience on Azure Cloud, Nginx Ingress doesn't fit to Che. My recommendation is to use Traefik.
I suggested in #12645 but have not published yet.

**Comment by dorinburcea** [comment]
I used charts and current deployment is AWS do you think is enough  to do a helm upgrade after changing the followings values 
1. values.yaml change the "ingressAnnotationPrefix" value to traefik 
![image](https://user-images.githubusercontent.com/21106256/54662887-7e32c480-4b1a-11e9-9d3f-a95feac0e08d.png)
2. the configmap.yaml
CHE_INFRA_KUBERNETES_INGRESS_ANNOTATIONS__JSON

is this the final solution ?

**Comment by monaka** [comment]
I've not tried to deploy Che into EKS, But...

In addition you pointed, it will be required to fix all `kubernetes.io/ingress.class: "nginx"`.
At least `templates/ingress.yaml`.
And also `custom-charts/che-jaeger/templates/ingress.yaml`, `custom-charts/che-keycloak/templates/ingress.yaml` if you deployed them.

Some cares about HTTPS certificate may be also required. (I'm not sure but probably)

**Comment by garagatyi** [comment]
As far as I remember it is how NginX works. When new routing rule gets added NginX has to reload and it disconnects existing clients. AFAIK HAProxy does the same. Since Traefik is designed for containers it can be more suitable for such a case.

**Comment by dorinburcea** [comment]
@monaka I did installed CHE 6.19.0 with traefik. The infra is on kubernetes in AWS and the traefik service is behind AWS ELB (TCP Layer 4) -> SSL terminate on traefik. 

The situation is like this :  
  1. Che  start ok there are no errors
  2. I can login to the CHE dashboardand create new workspace. 
  3. I click start the workspace and I observed this. 
              a) All ingress are created 
              b) Workspace pod is created immediatly and running in kubernetes and I can open terminal in the pod and bootstrap.log is ok
              d) The dashboard instead doesn't trigger any websocket connection. 
              e) finally can't detect the event of workspace statusChange in the CHE backend i can see logs like this:

[DEBUG] [o.e.c.a.c.n.EventService 107]        - Publish event {"error":"Server \u0027wsagent/http\u0027 in machine \u0027dev-machine\u0027 not available.","status":"STOPPED","workspaceId":"workspace8fc6qhwsil4gtp7v","prevStatus":"STARTING"} for org.eclipse.che.api.workspace.activity.WorkspaceActivityManager$UpdateStatusChangedTimestampSubscriber@5b484ffd
2019-03-24 19:48:34,698[aceSharedPool-0]  
[DEBUG] [o.e.c.a.c.n.EventService 107]        - Publish event {"error":"Server \u0027wsagent/http\u0027 in machine \u0027dev-machine\u0027 not available.","status":"STOPPED","workspaceId":"workspace8fc6qhwsil4gtp7v","prevStatus":"STARTING"} for org.eclipse.che.api.core.notification.RemoteSubscriptionManager$$Lambda$61/492489374@58995c1c
2019-03-24 19:48:34,698[aceSharedPool-0]  
[DEBUG] [c.j.c.t.EndpointIdConfigurator 46]   - Configuring outgoing request endpoint ID: 1356602418<-:->master-websocket-endpoint
2019-03-24 19:48:34,698[aceSharedPool-0]  
[DEBUG] [c.j.c.t.MethodNameConfigurator 54]   - Configuring outgoing request method name name: workspace/statusChanged
2019-03-24 19:48:34,698[aceSharedPool-0]  
[DEBUG] [c.a.c.j.c.t.ParamsConfigurator 56]   - Configuring outgoing request params: endpoint ID: 1356602418<-:->master-websocket-endpoint, method: workspace/statusChanged, params object class: class org.eclipse.che.api.workspace.server.dto.DtoServerImpls$WorkspaceStatusEventImpl, params object value: {"error":"Server \u0027wsagent/http\u0027 in machine \u0027dev-machine\u0027 not available.","status":"STOPPED","workspaceId":"workspace8fc6qhwsil4gtp7v","prevStatus":"STARTING"}
2019-03-24 19:48:34,698[aceSharedPool-0]  
[DEBUG] [.j.c.t.SendConfiguratorFromOne 61]   - Transmitting request: endpoint ID: 1356602418<-:->master-websocket-endpoint, method: workspace/statusChanged, params object class: class org.eclipse.che.api.workspace.server.dto.DtoServerImpls$WorkspaceStatusEventImpl, params list value{"error":"Server \u0027wsagent/http\u0027 in machine \u0027dev-machine\u0027 not available.","status":"STOPPED","workspaceId":"workspace8fc6qhwsil4gtp7v","prevStatus":"STARTING"}
2019-03-24 19:48:34,698[aceSharedPool-0]
